db/migration/1729763649580-SetYAxisMinDefaultToZero.ts

```diff
@@ -0,0 +1,177 @@
+        // of the chart itself nor the indicator config have yAxis.min set
+        await queryRunner.query(`
+            -- sql
+            UPDATE chart_configs cc
+            JOIN charts c ON cc.id = c.configId
+            -- keep charts without parent indicator
+            LEFT JOIN charts_x_parents cxp ON cxp.chartId = c.id
+            LEFT JOIN variables v ON v.id = cxp.variableId
+            LEFT JOIN chart_configs cc_admin ON cc_admin.id = v.grapherConfigIdAdmin
+            SET
+                -- using JSON_MERGE_PATCH instead of JSON_SET in case yAxis doesn't exist
+                cc.patch = JSON_MERGE_PATCH(cc.patch, '{"yAxis":{"min":"auto"}}')
+            WHERE
+                cc.full ->> '$.type' = 'LineChart'
+                AND c.isInheritanceEnabled IS TRUE
+                AND cc.patch ->> '$.yAxis.min' IS NULL
+                AND (
+                    cc_admin.full IS NULL
+                    OR cc_admin.full ->> '$.yAxis.min' IS NULL
+                )
+        `)
+
+        // recompute all full configs (we need to update all charts
+        // since the defaultGrapherConfig has changed)
+        const charts = await queryRunner.query(`
+             -- sql
+             SELECT
+                 cc.id AS configId,
+                 cc.patch AS patchConfig,
+                 cc_etl.patch AS etlConfig,
+                 cc_admin.patch AS adminConfig,
+                 c.isInheritanceEnabled
+             FROM charts c
+             JOIN chart_configs cc ON cc.id = c.configId
+             LEFT JOIN charts_x_parents cxp ON cxp.chartId = c.id
+             LEFT JOIN variables v ON v.id = cxp.variableId
+             LEFT JOIN chart_configs cc_etl ON cc_etl.id = v.grapherConfigIdETL
+             LEFT JOIN chart_configs cc_admin ON cc_admin.id = v.grapherConfigIdAdmin
+         `)
+        for (const chart of charts) {
+            const patchConfig = JSON.parse(chart.patchConfig)
+
+            const etlConfig = chart.etlConfig
+                ? JSON.parse(chart.etlConfig)
+                : undefined
+            const adminConfig = chart.adminConfig
+                ? JSON.parse(chart.adminConfig)
+                : undefined
+
+            const fullConfig = mergeGrapherConfigs(
+                defaultGrapherConfig as any,
+                chart.isInheritanceEnabled ? etlConfig ?? {} : {},
+                chart.isInheritanceEnabled ? adminConfig ?? {} : {},
+                patchConfig
+            )
+
+            await queryRunner.query(
+                `
+                    -- sql
+                    UPDATE chart_configs cc
+                    SET cc.full = ?
+                    WHERE cc.id = ?
+                `,
+                [JSON.stringify(fullConfig), chart.configId]
+            )
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public async down(): Promise<void> {}
+}
+
+const defaultGrapherConfig = {
```

Comment:
Oh shoot, I copy-pasted the wrong config! This should just be the default config as it exists on this branch. I copied it over (instead of importing) because I realised that relying on the default config version of the code that is running the migration is a bit dangerous (because we really do need this specific version of the default config). Sorry for the confusion!

So, the default config we're working with doesn't have a default value for `yAxis.min`. Then, the gist of the migration is:
- If a chart relied on the default for `yAxis.min` being `auto`, add `yAxis.min=auto` to its patch config to make this explicit
- Then recompute the full config using the new default config
- The full config should thus be unchanged (but the patch config of a chart might have been changed)

Before reading your comment, I was testing an upstream PR upstream and I couldn't make sense of some of the chart diffs I was seeing. I'm pretty sure this is the reason – so thanks for spotting it! It probably would have taken ages to figure it out 🫠